src/job_description_feature_extraction.py

```diff
@@ -68,3 +70,49 @@ def get_rake_keywords(job_description):
         # save top 3 results, the rest is usually garbage
         keywords.append(rake.run(doc)[:3])
     return keywords
+
+
+def get_top_idf_features(job_description, k):
+    """
+    use TfIdf to extract top k keywords of corpus
+
+    :param job_description: data frame
+    :param k: number of
+    :return: list of k top features
+    """
+    # based on https://stackoverflow.com/questions/25217510/
+    tfidf_vectorizer, term_vector = _vectorize(job_description)
+    # use _build_idf_dict to get a {term: score} dictionary which was the whole point here
+    # but it makes the build fail because of unused variables so yea
+    # idf_dict = _build_idf_dict(tfidf_vectorizer)
+    idf_ranking = np.argsort(tfidf_vectorizer.idf_)[::-1]
+    features = tfidf_vectorizer.get_feature_names()
+    return [features[i] for i in idf_ranking[:k]]
+
+
+def _build_idf_dict(tfidf_vectorizer):
+    """
+    builds a dictionarry of all terms
+
+    :param tfidf_vectorizer
+    :return: dictrionary in form of {term: score}
+    """
+    idf_dict = {}
+    features = tfidf_vectorizer.get_feature_names()
+    # TODO: not sure if that's the most efficient way to do this..
+    for term, score in zip(features, tfidf_vectorizer.idf_):
+        idf_dict[term] = score
+    return idf_dict
+
+
+def _vectorize(job_description):
+    """
+    vectorize job_descriptoins using tfidf
+
+    :param job_description: data_frame
+    :return: (vectorizer, term_vector)
+    """
+    tfidf_vectorizer = TfidfVectorizer()
+    job_description_list = job_description[JOB_DESCRIPTION_FIELD].values.tolist()
+    term_vector = tfidf_vectorizer.fit_transform(job_description_list)
+    return tfidf_vectorizer, term_vector
```

Comment:
problem here is, that we need both later on, the vectorizer contains meta information about the fit_transformed data